### ISSUE TITLE
"'$this' can not be used in static closures" lint does not work for arrow functions

### ISSUE THREAD

**Issue by chriscpty** [kalessil/phpinspectionsea#1983]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP8.1                     |

Current behaviour
--
When using `$this` in a static closure, a warning is shown:
```php
$foo = static function () { return $this; };
                                   ^^^^^ [EA] '$this' can not be used in static closures.
```

The same warning does not show up for the arrow function syntax:

```
php
$foo = static fn () => $this; // no warning here
```

Expected behaviour
--
The lint should behave the same for both `function` and `fn` closures.
